### PR TITLE
New CardLink Component

### LINKED ISSUE #504: Develop the Clickable tile component
_hahnrob created the following on Oct 04:_  

 

#### User Story

> As a `[user role below]`:
As a developer using the Digital Design System

> I need to:
have a Clickable tile that can be used in the Locale Selector

> so that I can:
include this component in the Locale Selector in the web site I am developing

#### Additional information

- This issue is dependent on the design and the [package architecture https://github.com/carbon-design-system/ibm-dotcom-library/issues/504](https://zenhub.ibm.com/app/workspaces/digital-design-system-5b3a815a8bf6a932efa6fd03/issues/webstandards/digital-design/860) 
- [final design and specs](https://ibm.box.com/s/isie20a89socbo9ael687u6pb75uz9f8)
              - Includes:  Functional labels, Visual redlines, Behavior and interaction specs, Desktop and mobile
 - [Box link to Mary O'Donnell's functional specs](https://ibm.ent.box.com/folder/88263129224)
 - While this clickable tile is included in the Locale Selector, it will be developed as a generic tile.
 - [Carbon clickable tile](https://www.carbondesignsystem.com/components/tile/code#clickable-tile)

#### Acceptance criteria
 - [ ] Component has been tested using Netlify
 - [ ] The component development follows a modular component pattern
 - [ ] The component should be accessible via ES6 import
 - [ ] The component should be visible within the living documentation of the Dotcom components
 - [ ] Presented and approved Product Owner

 

_Original issue: https://github.ibm.com/webstandards/digital-design/issues/1805_

### PR DESCRIPTION
**Pull request by photodow**
### Related Ticket(s)

Closes #504

### Description

This pull request creates a new `CardLink` component for the locale selector, card list, long form patterns, and more.

image

> Note: there is a known bug in this at the moment. Pulling in the `CardLink` by itself should be fine, but the way/order storybook is pulling in the `footer` with themed `g90` `bx--link` component it's overriding some of the styling since Carbon added the `.bx--link` class to the clickable tiles. I'm looking into how we can fix this, and will put in a follow up pull request.
>
> To test with proper styling pull down locally and temporarily comment out [this line](https://github.com/carbon-design-system/ibm-dotcom-library/blob/master/packages/styles/scss/components/link/_link-expressive.scss#L8)
> ```scss
> ...
> // @import '~carbon-components/scss/components/link/link';
> @import '../../globals/vars.scss';
> @import '../../globals/mixins.scss';
> ...
> ```

### Changelog

**New**

- CardLink component (behind feature flag)

### CODE REVIEW

File: packages/react/src/components/CardLink/CardLink.js [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+ *
+ * @param {object} props react proptypes
+ * @returns {object} JSX object
+ */
+const CardLink = ({ title, href, content, icon, className, ...props }) => {
+  if (!title || !href) {
+    return null;
+  }
+
+  console.log('testing');
```
Comment (jeffchew) [inline packages/react/src/components/CardLink/CardLink.js]: Stray console log here...

Comment (photodow) [reply packages/react/src/components/CardLink/CardLink.js]: Boom, thanks. I thought I got them all.

File: packages/react/src/components/CardLink/CardLink.js [hunk 2]
```diff
@@ -0,0 +1,86 @@
+/**
+ * Copyright IBM Corp. 2016, 2018
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import { CARD_LINK } from '../../internal/FeatureFlags.js';
```
Comment (jeffchew) [inline packages/react/src/components/CardLink/CardLink.js]: @photodow I just added `CARD_LINK` to react experimental, so we would need to kick off another preview build to see the card.

File: packages/react/src/components/CardLink/CardLink.js [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import { settings as ddsSettings } from '@carbon/ibmdotcom-utilities';
+import { settings } from 'carbon-components';
+import { ClickableTile } from 'carbon-components-react';
+
+const { stablePrefix } = ddsSettings;
+const { prefix } = settings;
+
+console.log('testing 1,2,3');
```
Comment (jeffchew) [inline packages/react/src/components/CardLink/CardLink.js]: Another stray console log...

File: packages/react/src/components/CardLink/CardLink.js (hunk 2 above)
Comment (photodow) [reply packages/react/src/components/CardLink/CardLink.js]: pushed another commit, so this should be done. let me know otherwise.

Comment (jeffchew) [reply packages/react/src/components/CardLink/CardLink.js]: There it is:
https://deploy-preview-554--ibmdotcom-react-experimental.netlify.com/?path=/story/card--link

**Comment by jeffchew** [comment]
@photodow visually it seems off in the preview, perhaps an import issue or something?

Screen Shot 2019-10-22 at 11 25 03 AM

**Comment by photodow** [comment]
@jeffchew I'm already aware of that issue, and figured since it's behind a feature flag, and @sannminwin needs it sooner rather then later I'll figure it out in a seperate PR. It appears to be less of an issue with the card itself and more with how storybook is building it, but I'm still looking into it. You can see my comment above in the description.

> *Note: there is a known bug in this at the moment. Pulling in the `CardLink` by itself should be fine, but the way/order storybook is pulling in the `footer` with themed `g90` `bx--link` component it's overriding some of the styling since Carbon added the `.bx--link` class to the clickable tiles. I'm looking into how we can fix this, and will put in a follow up pull request.*
>
> *To test with proper styling pull down locally and temporarily comment out [this line](https://github.com/carbon-design-system/ibm-dotcom-library/blob/master/packages/styles/scss/components/link/_link-expressive.scss#L8)*
> ```scss
> ...
> // @import '~carbon-components/scss/components/link/link';
> @import '../../globals/vars.scss';
> @import '../../globals/mixins.scss';
> ...
> ```

**Comment by jeffchew** [comment]
> @jeffchew I'm already aware of that issue, and figured since it's behind a feature flag, and @sannminwin needs it sooner rather then later I'll figure it out in a seperate PR. It appears to be less of an issue with the card itself and more with how storybook is building it, but I'm still looking into it. You can see my comment above in the description.

Issue created to track this: https://github.com/carbon-design-system/ibm-dotcom-library/issues/560

**Comment by jeffchew** [comment]
This has been merged. @sannminwin , please incorporate this into your branch.

**Pull request merged**